lib/shared/bucketing-assembly-script/assembly/types/target.ts

```diff
@@ -231,9 +234,35 @@ export class CustomDataFilter extends UserFilter {
         this.dataKey = getStringFromJSON(filter, 'dataKey')
         this.dataKeyType = isValidString(filter, 'dataKeyType', validDataKeyTypes)
     }
+    stringify(): string {
+        const json = new JSON.Obj()
+        if (this.dataKey) {
+            json.set('dataKey', this.dataKey)
+        }
+        if (this.dataKeyType) {
+            json.set('dataKeyType', this.dataKeyType)
+        }
+        if (this.type) {
+            json.set('type', this.type)
+        }
+        if (this.subType) {
+            json.set('subType', this.subType)
+        }
+        if (this.comparator) {
+            json.set('comparator', this.comparator)
+        }
+        if (this.values) {
+            json.set('values', this.values)
+        }
+        return json.stringify()
+    }
 }
 
-function initializeFilterClass(filter: JSON.Obj): AudienceFilterOrOperator {
+function initializeFilterClass(filter: JSON.Obj): AudienceFilter | null {
+    if (isValidStringOptional(filter, 'operator', validAudienceOperators, false)){
```

Comment:
dont think we want this here anymore?

Reply:
uhhh yeah, I can change so that this method doesn't get called if it's an operator instead. It makes more sense there anyway